procurement_purchase_requisition_generation/tests/test_procurement_purchase_requisition_generation.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2025 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields
+from odoo.tests import Form
+from odoo.tools import mute_logger
+
+from odoo.addons.base.tests.common import BaseCommon
+from odoo.addons.stock.tests import common
+
+
+class TestProcurementPurchaseRequisitionGeneration(BaseCommon, common.TestStockCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(
+            context=dict(
+                cls.env.context, test_procurement_purchase_requisition_generation=True
+            )
+        )
+        cls.mto = cls.env.ref("stock.route_warehouse0_mto")
+        cls.buy = cls.env.ref("purchase_stock.route_warehouse0_buy")
+        cls.vendor = cls.env["res.partner"].create({"name": "Test vendor"})
+        cls.productA.write(
+            {
+                "seller_ids": [(0, 0, {"partner_id": cls.vendor.id})],
+                "route_ids": [(6, 0, [cls.buy.id, cls.mto.id])],
+            }
+        )
+        cls.productB.write(
+            {
+                "route_ids": [(6, 0, [cls.buy.id, cls.mto.id])],
+            }
```

Comment:
Remember to not put trailing commas for avoiding extra lines:

```suggestion
            {"route_ids": [(6, 0, [cls.buy.id, cls.mto.id])]}
```